### ISSUE TITLE
Share screenshots into card attachment

### ISSUE THREAD

**Issue by reox** [stefan-niedermann/nextcloud-deck#516]
**Is your feature request related to a problem? Please describe.**

Might be related to #248 (Not sure if there is already an issue for that matter though, have not found one)
I'm using https://f-droid.org/en/packages/de.beatbrot.screenshotassistant/ to make screenshots. I would like to be able to attach them to cards directly using the share button.
However, If I click the share button I get the message "not supported yet".
Is it because the image is not stored on disk at the moment of sharing?

**Describe the solution you'd like**

Should be able to save transient images as attachment.

**Describe alternatives you've considered**

Storing them to disk is tedious, as with screenshot assistant you can crop the image in place and send it to the application you want (for example I use it to send screenshots to LectureNotes)

**Comment by stefan-niedermann** [comment]
yeah, i'm not sure how this can be implemented yet. Maybe we have to write incoming files to the disk ourself before uploading them.

Maybe you can contribute a PoC Pull Request @reox ? 🙂 🚀

**Comment by desperateCoder** [comment]
Keep in mind, that we also have offline capabilities. This makes the disk interaction kind of mandatory, I guess.

**Comment by reox** [comment]
I read that there is a attachment API: https://github.com/nextcloud/deck/blob/master/docs/API.md#post-boardsboardidstacksstackidcardscardidattachments---upload-an-attachment
So it would probably be easy to take the binary stream and just POST it?

I also looked up the code in the screenshot tool: https://github.com/beatbrot/ScreenshotAssistant/blob/master/app/src/main/java/de/beatbrot/screenshotassistant/ScreenshotActivityViewModel.kt#L57
It almost looks like that the file is already written to disk, however the folder seems to be deleted for every new screenshot.
In the log I only see this:
```
ActivityManager: START u0 {act=android.intent.action.SEND typ=image/jpeg flg=0xb080001 cmp=ComponentInfo{it.niedermann.nextcloud.deck/it.niedermann.nextcloud.deck.ui.SelectCardActivity}} from uid 10209
```
Not sure how this all works together...

**Comment by stefan-niedermann** [comment]
Oh, the screenshot tool is from my buddy @beatbrot ? maybe he can give us a hint about how to deal with the results of his app? 😉

**Comment by desperateCoder** [comment]
@reox the actual upload is already implemented and working in our app, including offline capabilities. The only challenge here is to store the file at some place our app has permissions to read from.

**Comment by reox** [comment]
Btw I tested to upload an image via the share button to my nextcloud directly, which works quite nicely

**Comment by stefan-niedermann** [comment]
> Btw I tested to upload an image via the share button to my nextcloud directly, which works quite nicely

Of course, it's a different app and is not directly related to the capabilities of the Deck Android App.

**Comment by stefan-niedermann** [comment]
The issue seems to be caused here: https://github.com/stefan-niedermann/nextcloud-deck/blob/master/app/src/main/java/it/niedermann/nextcloud/deck/ui/SelectCardActivity.java#L54

We can't determine a path for the shared image and therefore not upload it. This might be either related to missing permissions or maybe we simply don't receive a file uri but some kind of blob which we do have to store as a file first. Though my knowledge about this kind of file handling is limited, therefore: Pull Requests and concrete solutions are welcome, otherwise we need you to be patient so i can figure out how this can be solved. :slightly_smiling_face:

**Comment by reox** [comment]
> > Btw I tested to upload an image via the share button to my nextcloud directly, which works quite nicely
> 
> Of course, it's a different app and is not directly related to the capabilities of the Deck Android App.

What I meant is that it looks like the nextcloud app gets the correct filename etc from the screenshot tool.

> The issue seems to be caused here: https://github.com/stefan-niedermann/nextcloud-deck/blob/master/app/src/main/java/it/niedermann/nextcloud/deck/ui/SelectCardActivity.java#L54

Okay, so what I see is this:
```
I DeckLog : onCreate() (SelectCardActivity.java:47) → android.intent.action.SEND
I DeckLog : onCreate() (SelectCardActivity.java:48) → image/jpeg
E DeckLog : onCreate() (SelectCardActivity.java:62) -> java.lang.IllegalArgumentException: column '_data' does not exist. Available columns: []
E DeckLog : 	at android.database.AbstractCursor.getColumnIndexOrThrow(AbstractCursor.java:340)
E DeckLog : 	at android.database.CursorWrapper.getColumnIndexOrThrow(CursorWrapper.java:87)
E DeckLog : 	at it.niedermann.nextcloud.deck.util.FileUtils.getDataColumn(FileUtils.java:320)
E DeckLog : 	at it.niedermann.nextcloud.deck.util.FileUtils.getPath(FileUtils.java:165)
E DeckLog : 	at it.niedermann.nextcloud.deck.ui.SelectCardActivity.onCreate(SelectCardActivity.java:54)
E DeckLog : 	at android.app.Activity.performCreate(Activity.java:7340)
E DeckLog : 	at android.app.Activity.performCreate(Activity.java:7331)
E DeckLog : 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1275)
E DeckLog : 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3105)
E DeckLog : 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3268)
E DeckLog : 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
E DeckLog : 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
E DeckLog : 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
E DeckLog : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1959)
E DeckLog : 	at android.os.Handler.dispatchMessage(Handler.java:106)
E DeckLog : 	at android.os.Looper.loop(Looper.java:214)
E DeckLog : 	at android.app.ActivityThread.main(ActivityThread.java:7080)
E DeckLog : 	at java.lang.reflect.Method.invoke(Native Method)
E DeckLog : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
E DeckLog : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```

However, it does not fail because it can not read the file it looks :)

**Comment by stefan-niedermann** [comment]
@reox i have implemented a potential solution. Would you mind downloading and testing [this `apk` file](https://github.com/stefan-niedermann/nextcloud-deck/suites/822761595/artifacts/9098468)? (You can install it in parallel to your currently installed Deck app)

**Comment by stefan-niedermann** [comment]
Fix will be in `1.5.1` - Please open a new issue if the problem still exists with this version.

**Comment by reox** [comment]
The link gives me a 404 :(

**Comment by reox** [comment]
Tested it now and it works like a charm! thank you very much!

**Comment by stefan-niedermann** [comment]
Really happy to hear that :slightly_smiling_face:
